symengine/mul.cpp

```diff
@@ -63,7 +63,7 @@ bool Mul::is_canonical(const RCP<const Number> &coef,
         // e.g. x**2**y (={x**2:y}), which should be represented as x**(2y)
         //     (={x:2y})
         if (is_a<Pow>(*p.first))
-            return false;
+            return true;
```

Comment:
This is not the correct way. You should return false if `p.first` is a `Pow` and `p.second` is not an integer.